### PR TITLE
feat: parallelize feature for tcp

### PR DESCRIPTION
**Pull request by rjha-splunk**

### CODE REVIEW

**Review by ikheifets-splunk — APPROVED** [review]

File: package/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #work.  If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 
 
-ARG SYSLOGNG_VERSION=4.2.0
+ARG SYSLOGNG_VERSION=4.3.1
```
Comment (ikheifets-splunk) [inline package/Dockerfile:19]: It seems that we also need update docker file for `sc4s lite`

File: package/etc/syslog-ng.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@version:4.2
+@version:4.3
```
Comment (ikheifets-splunk) [inline package/etc/syslog-ng.conf:1]: probably 4.3.1

File: package/etc/conf.d/log_paths/2/lp_dest_alts_global/plugin.py [hunk 3]
```diff
@@ -71,5 +84,7 @@
         mode=m["mode"],
         filter=m["filter"],
         dtype=m["dtype"],
+        enable_parallelize=normalize_env_variable_input(f"SC4S_ENABLE_PARALLELIZE"),
+        parallelize_no_partitions=int(os.getenv(f"SC4S_PARALLELIZE_NO_PARTITION", 4)),
```
Comment (mstopa-splunk) [inline package/etc/conf.d/log_paths/2/lp_dest_alts_global/plugin.py:88]: Can we update the docs in this PR?

**Pull request merged**